doc/cug.tex

```diff
@@ -4551,6 +4489,380 @@ \subsubsection{Collapsible Families In Suite Graphs}
 \label{fig-namespaces}
 \end{figure}
 
+\subsection{Parameterized Tasks}
+\label{Parameterized Tasks}
+
+Cylc can automatically generate tasks and dependencies by expanding
+parameterized task names over lists of parameter values.  Uses include:
+\begin{myitemize}
+    \item generating an ensemble of similar model runs
+    \item generating chains of tasks to process similar datasets
+    \item replicating an entire workflow, or part thereof, over several runs
+    \item splitting a model run into steps or ``chunks`` without dynamic cycling
+\end{myitemize}
+
+{\em Note that this can be done with Jinja2 loops too (Section~\ref{Jinja2})
+    but parameterization is much cleaner: nested loops can seriously reduce
+the clarity of a suite definition.}
+
+\subsubsection{Parameter Expansion}
+
+Parameters values can be lists of strings, or integer ranges (both bounds included):
+\begin{lstlisting}
+[cylc]
+    [[parameters]]
+        obs = ship, buoy, plane
+        run = 1..5  # 1, 2, 3, 4, 5
+\end{lstlisting}
+
+Angle brackets denote parameterization.  For example,
+\begin{lstlisting}
+    model<run>  # for run = 1..2
+\end{lstlisting}
+expands to,
+\begin{lstlisting}
+    model_run1, model_run2
+\end{lstlisting}
+and,
+\begin{lstlisting}
+    proc<obs>  # for obs = ship, buoy, plane
+\end{lstlisting}
+expands to,
+\begin{lstlisting}
+    proc_ship, proc_buoy, proc_plane
+\end{lstlisting}
+For integer values the parameter name appears in the expanded task name, but
+for string values it doesn't.  These default expansion templates  can be
+overridden if necesasry:
+\begin{lstlisting}
+[cylc]
+    [[parameters]]
+        obs = ship, buoy, plane
+        run = 1..5
+    [[parameter templates]]
+        run = -R%(run)s  # make foo<run> expand to foo-R1 etc.
+\end{lstlisting}
+(See~\ref{RefParameterTemplates} for more on the string template syntax.)
+
+Any number of parameters can be used at once.  This parameterization,
+\begin{lstlisting}
+    model<run,obs>  # for run = 1..2 and obs = ship, buoy, plane
+\end{lstlisting}
+expands to these tasks names,
+\begin{lstlisting}
+    model_run1_ship, model_run1_buoy, model_run1_plane,
+    model_run2_ship, model_run2_buoy, model_run2_plane
+\end{lstlisting}
+
+\subsubsection{Passing Parameter Values To Tasks}
+
+Parameter values are passed as environment variables to tasks generated by
+parameter expansion.  For the example above, task \lstinline=model_run2_ship=
```

Comment:
Ha, that's a habit I learned in typing classes in high school (on *typewriters*!) .  The modern rule does seem to be one space after a period.  LaTeX ignores extra spaces in any case (but, interestingly, and it apparently typesets a longer space after a period, by default).